javaslang/src/main/java/javaslang/Value.java

```diff
@@ -516,7 +516,7 @@ default CharSeq toCharSeq() {
      */
     @SuppressWarnings("unchecked")
     @GwtIncompatible("reflection is not supported")
-    default T[] toJavaArray(Class<T> componentType) {
+    default T[] toJavaArray(Class<? super T> componentType) {
```

Comment:
I think the only solution would be to provide an array wrapper class, alla https://github.com/functionaljava/functionaljava/blob/master/core/src/main/java/fj/data/Array.java